cargo-dist/src/lib.rs

```diff
@@ -144,6 +152,88 @@ fn run_build_step(
     }
 }
 
+fn build_fake(dist_graph: &DistGraph, target: &BuildStep, manifest: &DistManifest) -> Result<()> {
+    match target {
+        // These two are the meat: don't actually run these at all, just
+        // fake them out
+        BuildStep::Generic(target) => build_fake_generic_artifacts(dist_graph, target),
+        BuildStep::Cargo(target) => build_fake_cargo_artifacts(dist_graph, target),
+        // Never run rustup
+        BuildStep::Rustup(_) => Ok(()),
+        // Copying files is fairly safe
+        BuildStep::CopyFile(CopyStep {
+            src_path,
+            dest_path,
+        }) => copy_file(src_path, dest_path),
+        BuildStep::CopyDir(CopyStep {
+            src_path,
+            dest_path,
+        }) => copy_dir(src_path, dest_path),
+        BuildStep::CopyFileOrDir(CopyStep {
+            src_path,
+            dest_path,
+        }) => copy_file_or_dir(src_path, dest_path),
+        // The remainder of these are mostly safe to run as fake steps
+        BuildStep::Zip(ZipDirStep {
+            src_path,
+            dest_path,
+            zip_style,
+            with_root,
+        }) => zip_dir(src_path, dest_path, zip_style, with_root.as_deref()),
+        BuildStep::GenerateInstaller(installer) => {
+            generate_installer(dist_graph, installer, manifest)
+        }
+        BuildStep::Checksum(ChecksumImpl {
+            checksum,
+            src_path,
+            dest_path,
+        }) => Ok(generate_and_write_checksum(checksum, src_path, dest_path)?),
+        // Except source tarballs, which are definitely not okay
+        BuildStep::GenerateSourceTarball(SourceTarballStep {
+            committish,
+            prefix,
+            target,
+        }) => Ok(generate_fake_source_tarball(
+            dist_graph, committish, prefix, target,
+        )?),
+        // Or extra artifacts, which may involve real builds
+        BuildStep::Extra(target) => run_fake_extra_artifacts_build(dist_graph, target),
+    }
+}
+
+fn build_fake_cargo_artifacts(dist: &DistGraph, target: &CargoBuildStep) -> Result<()> {
+    generate_fake_artifacts(dist, &target.expected_binaries)
+}
+
+fn build_fake_generic_artifacts(dist: &DistGraph, target: &GenericBuildStep) -> Result<()> {
+    generate_fake_artifacts(dist, &target.expected_binaries)
+}
+
+fn run_fake_extra_artifacts_build(dist: &DistGraph, target: &ExtraBuildStep) -> Result<()> {
+    for artifact in &target.expected_artifacts {
+        let path = dist.dist_dir.join(artifact);
+        std::fs::File::create(&path).into_diagnostic()?;
+    }
+
+    Ok(())
+}
+
+fn generate_fake_artifacts(dist: &DistGraph, binaries: &[BinaryIdx]) -> Result<()> {
+    if !dist.target_dir.exists() {
+        std::fs::create_dir(&dist.target_dir).into_diagnostic()?;
+    }
+
+    for idx in binaries {
+        let binary = dist.binary(*idx);
+
+        for exe_dest in &binary.copy_exe_to {
+            std::fs::File::create(exe_dest).into_diagnostic()?;
+        }
```

Comment:
we never actually populate it today but there's `copy_symbols_to` or something as well, it would be nice to mock that as well for future proofing on us re-enabling it.

Reply:
Oh huh, I forgot about that one. OK, will add.